### PR TITLE
feature: add graphql codegen and update query hooks

### LINKED ISSUE #1025: GraphQL code generator
**Description**
Use GraphQL code generator for typing

#### Definition of done
- [ ] Description is created (Task creator)
- [ ] Code is complete (Developer)
- [ ] Scope of test and what was done section is defined in task (Developer)
- [ ] Review passed (Code reviewer)
- [ ] Verified manually (QA)
- [ ] Test cases created/updated if necessary (QA)
- [ ] Change successfully merged
- [ ] Release notes added if needed

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Fixes #1025

Creates new hooks to fetch data from GraphQL with Apollo client. Also infers typing directly from GraphQL endpoint.

# To Test

- [ ] dApp still works as before

# Background

There are some queries regarding 24h volume that could be updated but will be deprecated when #1040 is fixed.

### CODE REVIEW

**Review by Mi-Lan — COMMENTED** [review]
This looks so much better! Nice job!
Here is mu suggestion so we dry up queries even more it would be good to use fragments for fields that are same and repeated. This is what I found and would suggest because it would clean up code a bit.

1) Have fragment for token 0 and token 1 query since they are almost always in this combination and are repeated 5 times throughout the queries.
2) Fragment for rewards query since its also repeated 5 times
3) I saw these properties repeated through the liqudityMininingCapmaing queries which can be like a base fragment for the them Address, Duration,StartsAt,Endsat,Locked,StakingCap, Staked amount
 Wdyt @berteotti

**Comment by berteotti** [comment]
> This looks so much better! Nice job! Here is mu suggestion so we dry up queries even more it would be good to use fragments for fields that are same and repeated. This is what I found and would suggest because it would clean up code a bit. All of these repeat at least 5 times.
> 
> 1. Have fragment for token 0 and token 1 query since they are almost always in this combination and are repeated 5 times throughout the queries.
> 2. Fragment for rewards query since its also repeated 5 times
> 3. I saw these properties repeated through the liqudityMininingCapmaing queries which can be like a base fragment for the them Address, Duration,StartsAt,Endsat,Locked,StakingCap, Staked amount
>    Wdyt @berteotti

Good suggestion @Mi-Lan, its now done in [5772196](https://github.com/levelkdev/swapr-dapp/pull/1326/commits/57721967258a8d94428f37702dfd3bc42180d44f). Tell me your thoughts!

File: src/graphql/DerivedNativeCurrencyTokens.query.gql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+query DerivedNativeCurrencyTokens($tokenIds: [ID!]!) {
```
Comment (adamazad) [inline src/graphql/DerivedNativeCurrencyTokens.query.gql:1]:
```suggestion
query getDerivedNativeCurrencyTokens($tokenIds: [ID!]!) {
```

File: src/graphql/LiquidityMiningCampaign.query.gql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+query LiquidityMiningCampaign($liquidityMiningCampaignId: ID!) {
```
Comment (adamazad) [inline src/graphql/LiquidityMiningCampaign.query.gql:1]:
```suggestion
query getLiquidityMiningCampaign($liquidityMiningCampaignId: ID!) {
```

File: src/graphql/LiquidityMiningCampaings.query.gql [hunk 3]
```diff
@@ -0,0 +1,21 @@
+query LiquidityMiningCampaigns($userId: String) {
```
Comment (adamazad) [inline src/graphql/LiquidityMiningCampaings.query.gql:1]:
```suggestion
query getUserLiquidityMiningCampaigns($userId: String) {
```

File: src/graphql/Pair.query.gql [hunk 4]
```diff
@@ -0,0 +1,12 @@
+query Pair($pairId: ID!) {
```
Comment (adamazad) [inline src/graphql/Pair.query.gql:1]:
```suggestion
query getPair($pairId: ID!) {
```

File: src/graphql/Pairs.query.gql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+query Pairs {
```
Comment (adamazad) [inline src/graphql/Pairs.query.gql:1]:
```suggestion
query getPairs {
```

File: src/graphql/SingleSidedStakingCampaign.query.gql [hunk 6]
```diff
@@ -0,0 +1,11 @@
+query SingleSidedStakingCampaign($campaignId: ID!) {
```
Comment (adamazad) [inline src/graphql/SingleSidedStakingCampaign.query.gql:1]:
```suggestion
query getSingleSidedStakingCampaign($campaignId: ID!) {
```

File: src/graphql/SingleSidedStakingCampaigns.query.gql [hunk 7]
```diff
@@ -0,0 +1,15 @@
+query SingleSidedStakingCampaigns($stakeTokenId: String, $userId: String) {
```
Comment (adamazad) [inline src/graphql/SingleSidedStakingCampaigns.query.gql:1]:
```suggestion
query getSingleSidedStakingCampaignsByStakeTokenIdAndUserId($stakeTokenId: String, $userId: String) {
```

File: src/graphql/StakingCampaings.query.gql [hunk 8]
```diff
@@ -0,0 +1,14 @@
+query StakingCampaigns($userId: String) {
```
Comment (adamazad) [inline src/graphql/StakingCampaings.query.gql:1]:
```suggestion
query getUserStakingCampaigns($userId: String) {
```

File: src/graphql/Token.query.gql [hunk 9]
```diff
@@ -0,0 +1,5 @@
+query Token($tokenId: ID!) {
```
Comment (adamazad) [inline src/graphql/Token.query.gql:1]:
```suggestion
query getToken($tokenId: ID!) {
```

File: src/graphql/UserLiquidityPositions.query.gql [hunk 10]
```diff
@@ -0,0 +1,44 @@
+query UserLiquidityPositions($userId: String, $endsAtLowerLimit: BigInt!) {
```
Comment (adamazad) [inline src/graphql/UserLiquidityPositions.query.gql:1]:
```suggestion
query getUserLiquidityPositions($userId: String, $endsAtLowerLimit: BigInt!) {
```

File: src/graphql/Bundle.query.gql [hunk 11]
```diff
@@ -0,0 +1,5 @@
+query Bundle {
```
Comment (adamazad) [inline src/graphql/Bundle.query.gql:1]:
```suggestion
query getBundle {
```

File: src/graphql/PairLiquidityMiningCampaings.query.gql [hunk 12]
```diff
@@ -0,0 +1,18 @@
+query PairLiquidityMiningCampaings($endsAtLowerLimit: BigInt!, $pairId: ID!) {
```
Comment (adamazad) [inline src/graphql/PairLiquidityMiningCampaings.query.gql:1]: Trying to think of a better name for this query 😅

```suggestion
query getPairLiquidityMiningCampaings($endsAtLowerLimit: BigInt!, $pairId: ID!) {
```

**Review by adamazad — APPROVED** [review]
Great job, Leo! Nicely done! Regarding my comments, feel free to ignore most of them; it's a personal preference. I like my queries to describe what they do :)

**Review by Diogomartf — APPROVED** [review]

**Comment by wixzi** [comment]
Great job @berteotti. 👏🏼 . I also prefer what @adamazad said in his comment. Start with `get` if you are getting data.

**Comment by berteotti** [comment]
@wixzi @adamazad Queries are now renamed in. [2bab1ca](https://github.com/levelkdev/swapr-dapp/pull/1326/commits/2bab1cad3e916ac6c5afda846d3bfebd841594a1). Will merge this if you guys agree 😄

**Comment by adamazad** [comment]
@berteotti fine by me.

**Review by Diogomartf — APPROVED** [review]

**Pull request merged**